### ISSUE TITLE
test_minidump_size_limit occasionally fails

### ISSUE THREAD

**Issue by afranchuk** [rust-minidump/minidump-writer#69]
The test fails sporadically on my local system (due to small differences). Based on the comments in the test, this isn't anything new.

I'm wondering why the logic/assertion [here](https://github.com/rust-minidump/minidump-writer/blob/bb371264f55e948afee3d45847877207d6465972/tests/linux_minidump_writer.rs#L354-L366) isn't simply `assert!(meta.len() <= minidump_size_limit)`. As far as I can tell that's what is logically being tested, and there isn't really any way to guarantee (without significantly more effort, I expect) that the minidumps are the same size (with an error tolerance). Alternatively, I don't see much harm in increasing the error tolerance such that it's much less likely to fail.

**Comment by gabrielesvelto** [comment]
Yes, I think that relaxing the assertion would be the way to go here. Enforcing a hard limit on the minidump size is too much of an effort for little value, this is really a soft limit.

**Comment by Jake-Shadle** [comment]
Yah that has been problematic in the past it's fine to relax it.

**Comment by afranchuk** [comment]
So to clarify, do we want to
1. `assert!(meta.len() <= minidump_size_limit)` - this is fairly permissive, given that it's allowing the minidump an extra 2MB over the prior run, and realistically _wouldn't_ sporadically fail, or
2. continue to instead assert that the minidump size is approximately the same as the prior run, but with a more tolerant setting of "approximately"?

**Comment by afranchuk** [comment]
So we [do test that](https://github.com/rust-minidump/minidump-writer/blob/main/tests/linux_minidump_writer.rs#L301). The unit test both tests that a larger limit won't be triggered and a smaller limit will. This issue is specifically with regard to the larger limit case (which makes it all the more frustrating that it sporadically fails, since logically it's a simpler case :smile:). I will adjust the larger limit case accordingly.
